quartz/components/scripts/search.inline.ts

```diff
@@ -71,20 +71,44 @@ function highlight(searchTerm: string, text: string, trim?: boolean) {
   }`
 }
 
+const p = new DOMParser()
 const encoder = (str: string) => str.toLowerCase().split(/([^a-z]|[^\x00-\x7F])/)
 let prevShortcutHandler: ((e: HTMLElementEventMap["keydown"]) => void) | undefined = undefined
-document.addEventListener("nav", async (e: unknown) => {
-  const currentSlug = (e as CustomEventMap["nav"]).detail.url
+
+let fetchContentCache: Record<string, Element[] | undefined> = {}
+
+document.addEventListener("nav", async (e: CustomEventMap["nav"]) => {
+  const currentSlug = e.detail.url
 
   const data = await fetchData
   const container = document.getElementById("search-container")
   const sidebar = container?.closest(".sidebar") as HTMLElement
   const searchIcon = document.getElementById("search-icon")
   const searchBar = document.getElementById("search-bar") as HTMLInputElement | null
-  const results = document.getElementById("results-container")
+  const searchLayout = document.getElementById("search-layout")
   const resultCards = document.getElementsByClassName("result-card")
   const idDataMap = Object.keys(data) as FullSlug[]
 
+  const enablePreview = searchLayout?.dataset?.preview === "true"
+  let preview: HTMLDivElement | undefined = undefined
+  const results = document.createElement("div")
+  results.id = "results-container"
+  if (searchLayout?.querySelector(`#${results.id}`) === null) {
+    searchLayout?.appendChild(results)
+  }
+  results.style.flexBasis = enablePreview ? "30%" : "100%"
+
+  preview = document.createElement("div")
+  preview.id = "preview-container"
+  preview.style.flexBasis = "70%"
+
+  if (!enablePreview) {
+    preview.style.display = "none"
+  }
+  if (searchLayout?.querySelector(`#${preview.id}`) === null) {
+    searchLayout?.appendChild(preview)
+  }
```

Comment:
can we just not make `preview` and `preview-container` if `enablePreview` is false? just curious if there was a specific reason to just `display: none` instead of not adding it to the DOM

Reply:
Yes we can. I think we can chang the display to flex if enablePreview, so we don't have to figure out the display on scss.

one moment brb.

Reply:
This is now done.